files/image_config/memory_statistics/memory_statistics_handler.py

```diff
@@ -0,0 +1,1877 @@
+            logger.log_debug(f"Sent response: {response_json}")
+        except Exception as e:
+            logger.log_error(f"Failed to send response: {e}")
+
+    def handle_connection(self, connection):
+        """Processes a single incoming socket connection.
+        
+        This method reads the request data from the client, decodes it from JSON,
+        and processes it using the command handler. It handles any exceptions,
+        sending an error response if needed, and closes the connection afterward.
+        
+        :param connection: The socket connection established with the client.
+        """
+        error_response = {"status": False, "msg": None}
+        try:
+            request_data = connection.recv(4096)
+            if not request_data:
+                logger.log_warning("Received empty request")
+                return
+
+            request_json = json.loads(request_data.decode('utf-8'))
+            logger.log_debug(f"Received request: {request_json}")
+            command_name = request_json['command']
+            command_data = request_json.get('data', {})
+
+            response = self.command_handler(command_name, command_data)
+
+            self.send_response(connection, response)
+        except Exception as error:
+            logger.log_error(f"Error handling request: {traceback.format_exc()}")
+            error_response['msg'] = str(error)
+            self.send_response(connection, error_response)
+        finally:
+            try:
+                connection.close()
+                logger.log_debug("Connection closed")
+            except Exception as e:
+                logger.log_error(f"Error closing connection: {e}")
+
+    def stop_listening(self):
+        """Stops the socket listener loop by setting stop_event."""
+        logger.log_info("Stopping listener loop.")
+        self.stop_event.set() 
+        if self.listener_socket:
+            try:
+                self.listener_socket.close() 
+            except Exception as e:
+                logger.log_error(f"Error while closing listener socket: {e}")
+                           
+
+    def start_listening(self):
+        """Starts listening for incoming socket connections.
+        
+        This method initializes the socket, removes any existing socket files,
+        and enters a loop to accept and handle incoming connections. The loop
+        continues until the stop_event is set. Upon shutdown, it cleans up the
+        socket file and closes the listener socket.
+        """
+        self.safe_remove_file(self.address)
+        self.create_unix_socket()
+
+        while not self.stop_event.is_set():
+            try:
+                connection, client_address = self.listener_socket.accept()
+                logger.log_info("Accepted new connection")
+                self.handle_connection(connection)
+            except socket.timeout:
+                continue  
```

Comment:
>continue [](http://example.com/codeflow?start=16&length=8)

Add log in case timeout happen, but lost debuggability.

Reply:
Thanks for the feedback! I've added a logger.log_debug for socket.timeout to improve debuggability.